test/polyfill/media_capabilities_unit.js

```diff
@@ -172,5 +181,71 @@ describe('MediaCapabilities', () => {
       expect(requestKeySystemAccessSpy)
           .toHaveBeenCalledTimes(1);
     });
+
+    it('fails when the cast namespace is not available', async () => {
+      // Mock Shaka throwing an error from identifying an unexpected, non-Cast
+      // platform when performing Cast logic.
+      const isChromecastSpy =
+          spyOn(shaka['util']['Platform'],
+              'isChromecast').and.returnValue(true);
```

Comment:
Do spyOn calls automatically reset the method to the original after a test case?  I've always assumed they didn't, but this would be really useful if it works that way.

Reply:
It looks like it, based on https://github.com/jasmine/jasmine/issues/683#issuecomment-261629706.

Reply:
That's excellent.  I need to use spyOn more instead of jasmine.createSpy.  We spend too much effort restoring originals.